### PR TITLE
Move MSSearchWebpart to Protein module

### PR DESCRIPTION
**Pull request by labkey-adam**
#### Rationale
TargetedMS references the MSSearchWebpart webpart, so it needs to be part of the Protein module.

Potential Proteins view is useful for TargetedMS, so move its provider. Wire up the MS2-specific container condition iff that module is present.

Stop special casing (and double rendering) specific views in DoProteinSearchAction.

Move Lorikeet (JS spectrum view) to Protein module

#### Related Pull Requests
* https://github.com/LabKey/commonAssays/pull/771

### CODE REVIEW

File: ms2/test/src/org/labkey/test/tests/ms2/XTandemTest.java [hunk 1]
```diff
@@ -144,19 +144,16 @@ private void verifyComparePeptides()
         uncheckCheckbox(Locator.name("exactMatch"));
         clickButton("Search");
         assertElementPresent(Locator.linkContainingText(SAMPLE_BASE_NAME + " (test2)"));
-        clickAndWait(Locator.id("expandCollapse-ProteinSearchProteinMatches"), 0);
```
Comment (labkey-jeckels) [inline ms2/test/src/org/labkey/test/tests/ms2/XTandemTest.java]: I'd like to leave this collapsed by default. Users typically want to see the results associated with a protein, so they don't care as much about the proteins themselves. However, if they don't see any results, it's useful to check the protein list to see what is eligible to match against.

File: protein/src/org/labkey/protein/ProteinController.java [hunk 2]
```diff
@@ -271,6 +264,73 @@ public void addNavTrail(NavTree root)
...
+                        if (form.isRestrictProteins())
+                        {
+                            sequencesTableInfo.addCondition(containerCondition);
+                        }
+                    }
+                    proteinsView.setTitle("Matching Proteins (" + (seqIds.length == 0 ? "None" : seqIds.length) + ")");
+                }
+            }
+
+            return proteinsView;
```
Comment (labkey-jeckels) [inline protein/src/org/labkey/protein/ProteinController.java:326]: I think we can keep the collapse behavior consistent by adding:
```
proteinsView.enableExpandCollapse("ProteinSearchProteinMatches", true);
```
though I haven't tested that.

**Review by labkey-jeckels — APPROVED** [review]
See request related to the collapsed state for one of the web parts

File: protein/src/org/labkey/protein/ProteinController.java (hunk 2 above)
Comment (labkey-adam) [reply protein/src/org/labkey/protein/ProteinController.java:326]: Yeah, that works

**Pull request merged**